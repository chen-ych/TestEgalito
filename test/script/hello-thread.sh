#!/bin/bash
mkdir -p tmp

ln -sf ../../src/libegalito.so
LD_LIBRARY_PATH=../../src EGALITO_DEBUG=/dev/null ../../src/loader \
	../binary/build/thread \
    >tmp/thread.out 2>&1
rm libegalito.so

if [ -z "$(diff hello-thread.expected tmp/thread.out)" ]; then
    echo "test passed"
else
    echo "test failed!"
    exit 1
fi
