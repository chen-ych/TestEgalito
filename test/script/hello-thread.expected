will reach a barrier
will reach a barrier
Hello, World!
Hello, World!
joining
